cleverhans/experimental/certification/optimization.py

```diff
@@ -171,7 +260,7 @@ def run_one_step(self, eig_init_vec_val, eig_num_iter_val,
     # Project onto feasible set of dual variables
     if self.current_step % self.params['projection_steps'] == 0:
 
-      projected_certificate = self.dual_object.project_dual()
+      projected_certificate = self.projected_dual_object.compute_certificate()
```

Comment:
`self.projected_dual_object.compute_certificate()` will add a few more ops to the TesnorFlow graph. It means that additional ops will be added to TensorFlow graph on each step because `run_one_step` is called on every step.

Is it possible to do this only once instead of every step?